spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/views/ResolvedV2View.scala

```diff
@@ -22,10 +22,12 @@ package org.apache.spark.sql.catalyst.plans.logical.views
 import org.apache.spark.sql.catalyst.analysis.LeafNodeWithoutStats
 import org.apache.spark.sql.catalyst.expressions.Attribute
 import org.apache.spark.sql.connector.catalog.Identifier
+import org.apache.spark.sql.connector.catalog.View
 import org.apache.spark.sql.connector.catalog.ViewCatalog
 
 case class ResolvedV2View(
   catalog: ViewCatalog,
-  identifier: Identifier) extends LeafNodeWithoutStats {
+  identifier: Identifier,
+  view: View) extends LeafNodeWithoutStats {
```

Comment:
I'm not a fan of keeping the view around like this. It's only used in one case where it shouldn't be because the check should be done in query execution rather than in job planning. I don't think this change is needed so I'd roll it back.